internal/cli/stream.go

```diff
@@ -0,0 +1,102 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/windmilleng/tilt/internal/stream"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/spf13/cobra"
+)
+
+type streamCmd struct {
+}
+
+func (c *streamCmd) register() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "stream",
+		Short: "display streaming tilt output",
+	}
+
+	return cmd
+}
+
+// blocks until we have a successful connection established
+func connect(ctx context.Context) (<-chan stream.MessageOrError, func() error, error) {
+	cli := stream.NewStreamClient()
+	ch, err := cli.Connect(ctx)
+
+	if err != nil {
+		log.Printf("Waiting for `tilt up` to start.")
+
+		for {
+			cli := stream.NewStreamClient()
+			ch, err = cli.Connect(ctx)
+			if err == nil {
+				break
+			}
+			select {
+			case <-ctx.Done():
+				return nil, nil, ctx.Err()
+			case <-time.After(time.Second):
+			}
+		}
+	}
+
+	return ch, cli.Close, err
+}
+
+func (c *streamCmd) run(ctx context.Context, args []string) error {
+	analyticsService.Incr("cmd.stream", nil)
+	defer analyticsService.Flush(time.Second)
+
+	span, ctx := opentracing.StartSpanFromContext(ctx, "stream")
+	defer span.Finish()
+
+	for {
+		ch, closer, err := connect(ctx)
+		if err != nil {
+			if err == context.Canceled {
```

Comment:
we've got this check in a few places, shouldn't we just check [here instead](http://bit.ly/2PkTwNW)? (Doesn't need to happen now)